### DISCUSSION TITLE
Extending time dimension

### DISCUSSION THREAD

**Discussion by dirkliebich** [atoti/atoti#501] [Q&A] [answered]
I have historical sales data into the cube and worth that added a date (time) dimension.

I created projections into the future for every product and now would like to see product history and the projection as a line chart.

How can I extend the future dates to the cube?

**Comment by bloussou** [comment]
Hello,
What do you mean by "extend the future dates to the cube" ? Is it adding data to your existing hierarchies ?

**Reply by dirkliebich** [reply]
I created forecasts for my timeseries, which go beyond the historical dates I had already loaded.

However, when I try bring the forecast table in I cannot add the new values using **date** as a key. The system tells me that date is already used.... (Which is correct, it's used in history)

How can I join history and forecast and use DATE for both values?

**Reply by Acriss** [reply]
Oh I think I get it.
You have an existing DATE hierarchy that goes back in the past, with value.
You're adding forecasts, and want these future dates to be merged into the DATE hierarchy that already exists, to have a single timeline.
Correct ?

**Reply by dirkliebich** [reply]
Exactly!

**Comment by bloussou** [comment, marked as answer]
Hello,
You can add your new data to the table using [table.load_pandas()](https://docs.atoti.io/latest/lib/atoti.table.html#atoti.table.Table.load_pandas) for instance. The date hierarchy will be automatically updated with the added rows. Can this work for you ?

**Reply by dirkliebich** [reply]
I'll try out and report back. Thanks so much!

**Reply by Acriss** [reply]
Hey!
Did it end up working correctly for you?
Cheers
